docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,300 @@
+
+```js
+<div id={'blorp'} />
+```
+
+transpiles into:
+
+```js
+React.createElement('div', {id: 'blorp'}, null)
+```
+
+If you want to test out how some specific JSX is converted into JavaScript, you can try out [the online Babel transpiler](http://tinyurl.com/j26hzbz).
+
+## HTML Tags and React Components
+
+The first part of a JSX tag determines the type of the JSX element.
+
+Capitalized types indicate that the JSX tag is referring to a React component. These tags get transpiled into a direct reference to the named variable, so if you use the JSX `<Foo />` expression, `Foo` must be in scope.
+
+Since JSX transpiles into calls to `React.createElement`, the `React` library must also always be in scope from your JSX code.
+
+For example, both of the imports are necessary in this code, even though 'React' and 'MyComponent' are not directly referenced from JavaScript:
+
+```js
+import React from 'react';
+import MyComponent from './MyComponent';
+
+export default function() {
+  return <MyComponent />;
+}
+```
+
+You can also refer to a React component using dot-notation from within JSX. This is convenient if you have a single module that exports many React components. For example, if you have a `Settings` module that exports a `Settings.DatePicker` component, you can invoke it directly from JSX with:
+
+```js
+import React from 'react';
+import Settings from './settings';
+
+export default function() {
+  return <Settings.DatePicker barColor={'blue'} />;
+}
+```
+
+You can do the same thing by assigning `DatePicker` to its own variable:
+
+```js
+import React from 'react';
+import Settings from './settings';
+
+var DatePicker = Settings.DatePicker;
+
+export default function() {
+  return <DatePicker barColor={'blue'} />;
+}
+```
+
+If the element type starts with a lowercase letter, React will create a HTML element. In general, name your components starting with capital letters, and this will be fine. If you do have a component that starts with a lowercase letter, assign it to a capitalized variable before using it in JSX.
+
+For example, this code will not run as expected:
+
+```js
+import React from 'react';
+
+function hello(props) {
+  // This use of <div> is legitimate because div is a valid HTML tag
+  return <div>Hello {props.toWhat}</div>;
+}
+
+export default function() {
+  // This code attempts to create an HTML <hello> tag and fails
+  return <hello {toWhat='World'} />
+}
+```
+
+You cannot use a general expression as the JSX element type. If you do want to use a general expression to indicate the type of the element, just assign it to a capitalized variable first. For example, if you have a `findComponent()` function that returns a component, and you want to render one:
+
+```js
+import React from 'react';
+import findComponent from './findComponent';
+
+function render1() {
+  // Not valid JSX
+  return <findComponent() />;
+}
+
+var MyComponent = findComponent();
```

Comment:
Let's move this inside the function? That would be equivalent to your first example then.